x-pack/plugins/fleet/common/types/rest_spec/agent.ts

```diff
@@ -62,6 +62,7 @@ export interface PostAgentCheckinResponse {
 export interface PostAgentEnrollRequest {
   body: {
     type: AgentType;
+    // TODO deprecated should be removed in 8.0.0
```

Comment:
can you add info about this to https://github.com/elastic/obs-dc-team/issues/334? a lot of other places will need to be changed for full removal (saved object mappings, server schemas, openapi spec, dev docs..)

Reply:
@jen-huang I improved a little the PR by removing the field from saved object and added a migration, also added info in the issue

Reply:
nice, thanks for that!